.github/workflows/ci.yml

```diff
@@ -266,3 +266,102 @@ jobs:
         with:
           name: fluvio-k8-logs
           path: /tmp/flv_sc.log
+
+  upgrade_test:
+    name: Upgrade cluster test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust: [stable]
+        include:
+          - os: ubuntu-latest
+            sccache-path: /home/runner/.cache/sccache
+    env:
+      RUST_BACKTRACE: full
+      RUSTC_WRAPPER: sccache
+      RUSTV: ${{ matrix.rust }}
+      SCCACHE_CACHE_SIZE: 2G
+      SCCACHE_DIR: ${{ matrix.sccache-path }}
+      # SCCACHE_RECACHE: 1 # Uncomment this to clear cache, then comment it back out
+      FLUVIO_CMD: true
+      FLV_SOCKET_WAIT: 600
+      FLV_CLUSTER_MAX_SC_VERSION_LOOP: 120
+      FLV_CLUSTER_MAX_SC_NETWORK_LOOP: 60
+      FLV_TEST_CONSUMER_WAIT: 300000
+    steps:
+      - uses: actions/checkout@v2
+      - run: helm version
+      - name: Install ${{ matrix.rust }}
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+      - name: Setup Minikube
+        uses: manusa/actions-setup-minikube@v2.3.1
+        with:
+          minikube version: 'v1.16.0'
+          kubernetes version: 'v1.19.2'
+      - name: Install sccache (ubuntu-latest)
+        if: matrix.os == 'ubuntu-latest'
+        env:
+          LINK: https://github.com/mozilla/sccache/releases/download
+          SCCACHE_VERSION: 0.2.13
+        run: |
+          SCCACHE_FILE=sccache-$SCCACHE_VERSION-x86_64-unknown-linux-musl
+          mkdir -p $HOME/.local/bin
+          curl -L "$LINK/$SCCACHE_VERSION/$SCCACHE_FILE.tar.gz" | tar xz
+          mv -f $SCCACHE_FILE/sccache $HOME/.local/bin/sccache
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - name: Install sccache (macos-latest)
+        if: matrix.os == 'macos-latest'
+        run: brew install sccache
+      - name: Install Rust ${{ matrix.rust }}
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          profile: minimal
+          override: true
+      - name: Cache cargo registry
+        uses: actions/cache@v2
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+      - name: Save sccache
+        uses: actions/cache@v2
+        continue-on-error: false
+        with:
+          path: ${{ matrix.sccache-path }}
+          key: ${{ runner.os }}-sccache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-sccache-
+      - name: Start sccache server
+        run: sccache --start-server
+      - name: Build
+        run: |
+          cargo build --release --bin fluvio 
+      - name: Make image
```

Comment:
We should separate out two types of fluvio cluster.  For now, `local` should be sufficient to test upgrade

Reply:
Can you even upgrade a local cluster? I ran into a few problems trying to avoid compilation and kubernetes.

1. Prerelease versions don't work 

```
$ fluvio version
Fluvio CLI           : 0.7.3
Fluvio CLI SHA256    : 6ff4969b99a67ee9d8f594475e1884f51313dd99ac5414c4a13ab797845a590f
Fluvio Platform      : 0.7.3 (local)
Git Commit           : 895c07f66347d089c2b85a9df28ed1a4b71f3cb3
OS Details           : Ubuntu 20.10 (kernel 5.8.0-45-generic)
=== Plugin Versions ===
Fluvio Cloud CLI (fluvio-cloud) : 0.1.5

$ curl -fsS https://packages.fluvio.io/v1/install.sh | VERSION=latest bash
fluvio: ⏳ Downloading Fluvio 0.7.4-alpha.0+f0b6236a6406914140b3dc92225f9f0208d818b5 for x86_64-unknown-linux-musl...

$ fluvio cluster upgrade --local
Performing pre-flight checks
✅ ok: Supported helm version is installed
✅ ok: Supported kubernetes version is installed
✅ ok: Kubernetes config is loadable
✅ ok: Fluvio system charts are installed
Error:
   0: Fluvio cluster error
   1: Fluvio cluster error
   2: Failed to install Fluvio locally
   3: Kubernetes client error
   4: client error: 409 Conflict
```

2. Older version local cluster upgrade don't work
```
$ fluvio version
Fluvio CLI           : 0.7.3
Fluvio CLI SHA256    : 6ff4969b99a67ee9d8f594475e1884f51313dd99ac5414c4a13ab797845a590f
Fluvio Platform      : Not available (local)
Git Commit           : 895c07f66347d089c2b85a9df28ed1a4b71f3cb3
OS Details           : Ubuntu 20.10 (kernel 5.8.0-45-generic)
=== Plugin Versions ===
Fluvio Cloud CLI (fluvio-cloud) : 0.1.5

$ fluvio cluster upgrade --local
Performing pre-flight checks
✅ ok: Supported helm version is installed
✅ ok: Supported kubernetes version is installed
✅ ok: Kubernetes config is loadable
✅ ok: Fluvio system charts are installed
Error:
   0: Fluvio cluster error
   1: Fluvio cluster error
   2: Failed to install Fluvio locally
   3: Kubernetes client error
   4: client error: 409 Conflict
```

3. Stable to stable local upgrades don't work
```
$ fluvio version
Fluvio CLI           : 0.7.3
Fluvio CLI SHA256    : 6ff4969b99a67ee9d8f594475e1884f51313dd99ac5414c4a13ab797845a590f
Fluvio Platform      : 0.7.3 (local)
Git Commit           : 895c07f66347d089c2b85a9df28ed1a4b71f3cb3
OS Details           : Ubuntu 20.10 (kernel 5.8.0-45-generic)
=== Plugin Versions ===
Fluvio Cloud CLI (fluvio-cloud) : 0.1.5

$ fluvio cluster upgrade --local
Performing pre-flight checks
✅ ok: Supported helm version is installed
✅ ok: Supported kubernetes version is installed
✅ ok: Kubernetes config is loadable
✅ ok: Fluvio system charts are installed
Error:
   0: Fluvio cluster error
   1: Fluvio cluster error
   2: Failed to install Fluvio locally
   3: Kubernetes client error
   4: client error: 409 Conflict
```

The manner I've got this job set up is the only way I've been able to succeed in upgrading in CI.

I start a minikube cluster with the newest release version. Then I build and upgrade from the `master` branch, which is expected to typically be newer than the stable version, but possibly a pre-release so I have to use `cluster upgrade --develop` and build the minikube image too.

Am I overthinking this?

Reply:
I removed the hourly tests from this PR, but it will require this one to be merged first.